lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -273,8 +274,13 @@ class CFunctionSignatureTypePrinter
                  bool isInOutParam) {
     auto *cd = CT->getDecl();
     if (cd->hasClangNode()) {
-      ClangSyntaxPrinter(os).printClangTypeReference(cd->getClangDecl());
-      os << " *"
+      const auto *clangDecl = cd->getClangDecl();
+      ClangSyntaxPrinter(os).printClangTypeReference(clangDecl);
+      bool alreadyPointer = false;
+      if (const auto *typedefDecl = dyn_cast<clang::TypedefNameDecl>(clangDecl))
+        if (importer::isCFTypeDecl(typedefDecl) && typedefDecl->getName().ends_with("Ref"))
```

Comment:
Do we need a "-Ref" suffix check here? Looks like `isCFTypeDecl` would return false for types that aren't pointers.

Reply:
Oh, indeed! Good catch!